tests/prague/eip7692_eof_v1/eip7676_address_space_extension/test_ase_opcodes.py

```diff
@@ -0,0 +1,262 @@
+"""
+Execution of CALLF, RETF opcodes within EOF V1 containers tests
+"""
+
+import pytest
+
+from ethereum_test_tools import (
+    Account,
+    Address,
+    Environment,
+    StateTestFiller,
+    TestAddress,
+    Transaction,
+)
+from ethereum_test_tools.common.conversions import BytesConvertible, to_bytes
+from ethereum_test_tools.eof.v1 import Container, Section
+from ethereum_test_tools.eof.v1.constants import NON_RETURNING_SECTION
+from ethereum_test_tools.vm.opcode import Opcodes as Op
+
+from .spec import EOF_FORK_NAME
+
+REFERENCE_SPEC_GIT_PATH = "EIPS/eip-7676.md"
+REFERENCE_SPEC_VERSION = (
+    "00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
+)
+
+pytestmark = pytest.mark.valid_from(EOF_FORK_NAME)
+
+# ID_EOF_BALANCE = 0
+# ID_BALANCE = 1
+ID_EXTCALL = 2
+ID_CALL = 3
+ID_EXTDELEGATECALL = 4
+ID_DELEGATECALL = 5
+ID_EXTSTATICCALL = 6
+ID_STATICCALL = 7
+ID_CALLCODE = 8
+
+BASE_ADDRESS = 100000000
+# ADDRESS_EOF_BALANCE = Address(BASE_ADDRESS + ID_EOF_BALANCE)
+# ADDRESS_BALANCE = Address(BASE_ADDRESS + ID_BALANCE)
+ADDRESS_EXTCALL = Address(BASE_ADDRESS + ID_EXTCALL)
+ADDRESS_CALL = Address(BASE_ADDRESS + ID_CALL)
+ADDRESS_EXTDELEGATECALL = Address(BASE_ADDRESS + ID_EXTDELEGATECALL)
+ADDRESS_DELEGATECALL = Address(BASE_ADDRESS + ID_DELEGATECALL)
+ADDRESS_EXTSTATICCALL = Address(BASE_ADDRESS + ID_EXTSTATICCALL)
+ADDRESS_STATICCALL = Address(BASE_ADDRESS + ID_STATICCALL)
+ADDRESS_CALLCODE = Address(BASE_ADDRESS + ID_CALLCODE)
```

Comment:
I'm not super excited about this pattern: my concern is regarding a potential change that we will introduce in the close future where the test function "requests" the pre-alloc to place a smart contract "somewhere" and return the address, like so:
```python
def test_function(state_test: StateTestFiller, pre: Alloc, ...):
    my_contract_address = pre.deploy_contract(Op.SSTORE(0, 1))
```
So in the future you cannot request a specific address as easily as now, so I think we should prepare for that by making changes to patterns like this.

Reply:
I'm not exactly sure what is being asked for as `Alloc` does not have the `deploy_contract` method right now,
but I did migrate the allocations to use `itertools.Count` to communicate it's looking for "whatever address is handed out".  This breaks ordering assumptions implicit in the BASE+ID code that existed before.

Do I need to move this into the test method instead of at top level?